apps/namadillo/src/atoms/integrations/services.ts

```diff
@@ -176,6 +177,77 @@ export const queryAndStoreRpc = async <T>(
   }
 };
 
+const isTransferLocalTimeout = (
+  tx: TransferTransactionData,
+  timeoutInMinutes = 10
+): boolean => {
+  return differenceInMinutes(Date.now(), tx.createdAt) > timeoutInMinutes;
+};
+
+const dispatchTransactionEvent = (
+  eventName: string,
+  tx: IbcTransferTransactionData
+): void => {
+  window.dispatchEvent(
+    new CustomEvent(eventName, {
+      detail: { ...tx },
+    })
+  );
+};
+
+const handleTransactionSuccess = (
+  hash: string,
+  changeTransaction: (
+    hash: string,
+    update: Partial<IbcTransferTransactionData>
+  ) => void,
+  tx: IbcTransferTransactionData,
+  resultTxHash?: string
+): void => {
+  const update: Partial<IbcTransferTransactionData> = {
+    status: "success",
+    currentStep: TransferStep.Complete,
+    ...(resultTxHash && { resultTxHash }),
+  };
+  changeTransaction(hash, update);
+  dispatchTransactionEvent("IbcTransfer.Success", tx);
+};
+
+const handleTransactionError = (
+  hash: string,
+  changeTransaction: (
+    hash: string,
+    update: Partial<IbcTransferTransactionData>
+  ) => void,
+  tx: IbcTransferTransactionData,
+  errorMessage: string,
+  resultTxHash?: string
+): void => {
+  const update: Partial<IbcTransferTransactionData> = {
+    status: "error",
+    errorMessage,
+    ...(resultTxHash && { resultTxHash }),
+  };
+  changeTransaction(hash, update);
+  dispatchTransactionEvent("IbcTransfer.Error", tx);
+};
+
+const checkForIbcTransferTimeout = async (
+  client: StargateClient,
+  tx: IbcTransferTransactionData
+): Promise<[boolean, string]> => {
+  let isTimeout = isTransferLocalTimeout(tx);
+  let timeoutHash = "";
+
+  if (!isTimeout) {
+    const timeoutQuery = await queryForIbcTimeout(client, tx);
+    isTimeout = timeoutQuery.length > 0;
+    timeoutHash = timeoutQuery[0]?.hash || "";
+  }
+
+  return [isTimeout, timeoutHash];
```

Comment:
I'm thinking if we should use object on these cases. The array tuple is great for useState because it's a well know function and we rename the returned values. But, on other functions, it's harder to read and understanding the code, also, usually we don't rename the value, actually, we use the same name 
```tsx
const [isTimeout, timeoutHash] = await checkForIbcTransferTimeout(client, tx);
```